### ISSUE TITLE
[bde] build failure when python is python3

### ISSUE THREAD

**Issue by BillyONeal** [microsoft/vcpkg#18937]
**Host Environment**
- OS: Ubuntu 20.04
- Compiler: GCC 9

**To Reproduce**
`./vcpkg install bde`

**Failure logs** 
```
Building package bde[core]:x64-linux...
-- Using /home/bion/vcpkg/downloads/bloomberg-bde-tools-d4b1a7670829228f4ec81ecdccc598ce03ae8e80.tar.gz
-- Cleaning sources at /mnt/vcpkg-ci/buildtrees/bde/src/ce03ae8e80-ef877fdb7c.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/bion/vcpkg/downloads/bloomberg-bde-tools-d4b1a7670829228f4ec81ecdccc598ce03ae8e80.tar.gz
-- Using source at /mnt/vcpkg-ci/buildtrees/bde/src/ce03ae8e80-ef877fdb7c.clean
-- Configure bde-tools-v1.x
-- Using /home/bion/vcpkg/downloads/bloomberg-bde-3720d132d0879f19b9084cca62ebc75f1f24e1a3.tar.gz
-- Cleaning sources at /mnt/vcpkg-ci/buildtrees/bde/src/5f1f24e1a3-5405076f04.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/bion/vcpkg/downloads/bloomberg-bde-3720d132d0879f19b9084cca62ebc75f1f24e1a3.tar.gz
-- Using source at /mnt/vcpkg-ci/buildtrees/bde/src/5f1f24e1a3-5405076f04.clean
-- Configuring x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: waf configure --library-type=static --use-flat-include-dir --abi-bits=64 --prefix=/mnt/vcpkg-ci/packages/bde_x64-linux/debug --out=/mnt/vcpkg-ci/buildtrees/bde/x64-linux-dbg/bde-3.2.0.0 --build-type=debug
    Working Directory: /mnt/vcpkg-ci/buildtrees/bde/src/5f1f24e1a3-5405076f04.clean
    Error code: 2
    See logs for more information:
      /mnt/vcpkg-ci/buildtrees/bde/configure-x64-linux--dbg-out.log

Call Stack (most recent call first):
  ports/bde/portfile.cmake:67 (vcpkg_execute_required_process)
  scripts/ports.cmake:139 (include)


Error: Building package bde:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: bde:x64-linux
  Vcpkg version: 2021-05-05-unknownhash
```

```
Traceback (most recent call last):
  File "/mnt/vcpkg-ci/buildtrees/bde/src/ce03ae8e80-ef877fdb7c.clean/bin/.waf3-1.9.9-4469eb5c219e16263f641e0916ce6a48/waflib/Node.py", line 295, in ant_iter
    raise StopIteration
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/mnt/vcpkg-ci/buildtrees/bde/src/ce03ae8e80-ef877fdb7c.clean/bin/.waf3-1.9.9-4469eb5c219e16263f641e0916ce6a48/waflib/Scripting.py", line 120, in waf_entry_point
    run_commands()
  File "/mnt/vcpkg-ci/buildtrees/bde/src/ce03ae8e80-ef877fdb7c.clean/bin/.waf3-1.9.9-4469eb5c219e16263f641e0916ce6a48/waflib/Scripting.py", line 177, in run_commands
    parse_options()
  File "/mnt/vcpkg-ci/buildtrees/bde/src/ce03ae8e80-ef877fdb7c.clean/bin/.waf3-1.9.9-4469eb5c219e16263f641e0916ce6a48/waflib/Scripting.py", line 150, in parse_options
    Context.create_context('options').execute()
  File "/mnt/vcpkg-ci/buildtrees/bde/src/ce03ae8e80-ef877fdb7c.clean/bin/.waf3-1.9.9-4469eb5c219e16263f641e0916ce6a48/waflib/Options.py", line 149, in execute
    super(OptionsContext,self).execute()
  File "/mnt/vcpkg-ci/buildtrees/bde/src/ce03ae8e80-ef877fdb7c.clean/bin/.waf3-1.9.9-4469eb5c219e16263f641e0916ce6a48/waflib/Context.py", line 88, in execute
    self.recurse([os.path.dirname(g_module.root_path)])
  File "/mnt/vcpkg-ci/buildtrees/bde/src/ce03ae8e80-ef877fdb7c.clean/bin/.waf3-1.9.9-4469eb5c219e16263f641e0916ce6a48/waflib/Context.py", line 129, in recurse
    user_function(self)
  File "/mnt/vcpkg-ci/buildtrees/bde/src/5f1f24e1a3-5405076f04.clean/wscript", line 27, in options
    ctx.load('bdebuild.waf.wscript', tooldir=_get_tools_path(ctx))
  File "/mnt/vcpkg-ci/buildtrees/bde/src/ce03ae8e80-ef877fdb7c.clean/bin/.waf3-1.9.9-4469eb5c219e16263f641e0916ce6a48/waflib/Context.py", line 85, in load
    fun(self)
  File "/mnt/vcpkg-ci/buildtrees/bde/src/ce03ae8e80-ef877fdb7c.clean/lib/python/bdebuild/waf/wscript.py", line 73, in options
    ctx.load('compiler_c')
  File "/mnt/vcpkg-ci/buildtrees/bde/src/ce03ae8e80-ef877fdb7c.clean/bin/.waf3-1.9.9-4469eb5c219e16263f641e0916ce6a48/waflib/Context.py", line 85, in load
    fun(self)
  File "/mnt/vcpkg-ci/buildtrees/bde/src/ce03ae8e80-ef877fdb7c.clean/bin/.waf3-1.9.9-4469eb5c219e16263f641e0916ce6a48/waflib/Tools/compiler_c.py", line 40, in options
    opt.load_special_tools('c_*.py',ban=['c_dumbpreproc.py'])
  File "/mnt/vcpkg-ci/buildtrees/bde/src/ce03ae8e80-ef877fdb7c.clean/bin/.waf3-1.9.9-4469eb5c219e16263f641e0916ce6a48/waflib/Context.py", line 319, in load_special_tools
    lst=self.root.find_node(waf_dir).find_node('waflib/extras').ant_glob(var)
  File "/mnt/vcpkg-ci/buildtrees/bde/src/ce03ae8e80-ef877fdb7c.clean/bin/.waf3-1.9.9-4469eb5c219e16263f641e0916ce6a48/waflib/Node.py", line 344, in ant_glob
    ret=[x for x in self.ant_iter(accept=accept,pats=[to_pat(incl),to_pat(excl)],maxdepth=kw.get('maxdepth',25),dir=dir,src=src,remove=kw.get('remove',True))]
  File "/mnt/vcpkg-ci/buildtrees/bde/src/ce03ae8e80-ef877fdb7c.clean/bin/.waf3-1.9.9-4469eb5c219e16263f641e0916ce6a48/waflib/Node.py", line 344, in <listcomp>
    ret=[x for x in self.ant_iter(accept=accept,pats=[to_pat(incl),to_pat(excl)],maxdepth=kw.get('maxdepth',25),dir=dir,src=src,remove=kw.get('remove',True))]
RuntimeError: generator raised StopIteration

```

**Additional context**
Ubuntu 20.04 contains no out of the box way to make python mean python2.

**Comment by PhoebeHui** [comment]
From https://bloomberg.github.io/bde/library_information/build.html, it looks upstream recommands to use python3 instead.

**Comment by adamncasey** [comment]
This is probably fixed in #31644
